src/index.ts

```
@@ -267,30 +357,200 @@ export function withdrawalTransaction(
     sequence: timelock,
   });
 
+  // withdraw tx always has 1 output only
+  const estimatedFee = getEstimatedFee(feeRate, psbt.txInputs.length, 1);
+
   psbt.addOutput({
     address: withdrawalAddress,
-    value: tx.outs[outputIndex].value - withdrawalFee,
+    value: tx.outs[outputIndex].value - estimatedFee,
```

I think we should have dust checks here as well. However, given that there's a single input, we should just throw an error if this output has less value than dust.

---

make sense